curp/src/client.rs

```diff
@@ -454,6 +470,48 @@ where
         }
     }
 
+    /// Fetch Read state from leader
+    /// # Errors
+    ///   `ProposeError::EncodingError` encoding error met while deserializing the propose id
+    #[inline]
+    pub async fn fetch_read_state(&self, cmd: &C) -> Result<ReadState, ProposeError> {
```

Comment:
Can we add some IT test to cover this `fetch_read_state`?

Reply:
Added